cocotb/share/lib/fli/FliObjHdl.cpp

```diff
@@ -264,7 +265,8 @@ int FliLogicObjHdl::initialise(std::string &name, std::string &fq_name)
         m_enum_map[m_value_enum[i][1]] = i;  // enum is of the format 'U' or '0', etc.
     }
 
-    m_val_buff = (char*)malloc(m_num_elems+1);
+    size_t amt = static_cast<size_t>(m_num_elems) + 1;
```

Comment:
Can you change the type of `m_num_elements` to `size_t` instead?

Reply:
`m_num_elems` is usually `1` or the length of a vector whose bounds are 32-bit ints, so `int` (or better yet `int32_t`) makes the most sense. It would also create a bunch of warnings in other places, so nothing to gain from that.